src/app/modules/shared_models/collectibles_model.nim

```diff
@@ -68,12 +78,22 @@ QtObject:
   proc setIsFetching*(self: Model, value: bool) =
     if value == self.isFetching:
       return
-    if value:
-      self.appendLoadingItems()
-    else:
-      self.removeLoadingItems()
     self.isFetching = value
     self.isFetchingChanged()
+    self.checkLoadingItems()
+
+  proc isUpdatingChanged(self: Model) {.signal.}
+  proc getIsUpdating*(self: Model): bool {.slot.} =
+    self.isUpdating
+  QtProperty[bool] isUpdating:
+    read = getIsUpdating
+    notify = isUpdatingChanged
+  proc setIsUpdating*(self: Model, isUpdating: bool) {.slot.} =
```

Comment:
Just curious, why is this a slot and then not a `write` accessor to the property above?

Reply:
Nice catch, this doesn't need to be a slot